scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -2044,23 +2226,28 @@ def run( self ):
         try:
             # check to see if the user has manually specified a cluster name, and if so then
             # force enable 'Starbuck' features
-            if self.__container_checker and self.__container_checker.k8s_cache.get_cluster_name() is not None:
+            if self.__container_checker and self.__container_checker.get_cluster_name( self.__container_checker.k8s_cache ) is not None:
                 self._logger.log( scalyr_logging.DEBUG_LEVEL_1, "Cluster name detected, enabling k8s metric reporting and controller information" )
                 self.__include_controller_info = True
                 self.__report_k8s_metrics = self.__report_container_metrics
 
+            # disable controller info if k8s queries are disabled
+            if self.__k8s_disable_api_server:
+                self.__include_controller_info = False
+
             if self.__report_k8s_metrics:
 
                 runtime = None
                 if k8s_cache:
+                    # TODO(czerwin):  If the api server has been disabled, we need to decide what to do here.
```

Comment:
Is this TODO still relevant?

Reply:
Yes.. if we turn on `report_k8s_metric`, then this part of the code will fail because `k8s_cache` is None if the api server is disabled.

Reply:
We can do this later.

Reply:
Filed bug https://scalyr.myjetbrains.com/youtrack/issue/AGENT-150